sdk/servicebus/azure-servicebus/tests/test_connection_string_parser.py

```diff
@@ -0,0 +1,44 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+
+import os
+import pytest
+from azure.servicebus import (
+    ServiceBusClient,
+    Message,
+    ServiceBusConnectionStringProperties,
+    ServiceBusConnectionStringParser,
+)
+
+from devtools_testutils import AzureMgmtTestCase
+from azure.servicebus._base_handler import ServiceBusSharedKeyCredential
+
+class ServiceBusConnectionStringParserTests(AzureMgmtTestCase):
+    def test_sb_conn_str_parse_cs(self, **kwargs):
+        conn_str = 'Endpoint=sb://resourcename.servicebus.windows.net/;SharedAccessKeyName=test;SharedAccessKey=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        parse_result = ServiceBusConnectionStringParser(conn_str).parse()
+        assert parse_result.endpoint == 'sb://resourcename.servicebus.windows.net/'
+        assert parse_result.fully_qualified_namespace == 'resourcename.servicebus.windows.net'
+        assert parse_result.shared_access_key_name == 'test'
+        assert parse_result.shared_access_key == 'THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
```

Comment:
You deserve to throw something heavy at me for how annoying this comment will be, but... Probably worth having a flow for the SAS token path as well.

Reply:
valid comment ;)